app/Http/Middleware/RequireTwoFactorAuthentication.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use App\Exceptions\Http\TwoFactorAuthRequiredException;
+use App\Filament\Pages\Auth\EditProfile;
+use App\Livewire\AlertBanner;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Str;
+
+class RequireTwoFactorAuthentication
+{
+    public const LEVEL_NONE = 0;
+
+    public const LEVEL_ADMIN = 1;
+
+    public const LEVEL_ALL = 2;
+
+    /**
+     * Check the user state on the incoming request to determine if they should be allowed to
+     * proceed or not. This checks if the Panel is configured to require 2FA on an account in
+     * order to perform actions. If so, we check the level at which it is required (all users
+     * or just admins) and then check if the user has enabled it for their account.
+     *
+     * @throws \App\Exceptions\Http\TwoFactorAuthRequiredException
+     */
+    public function handle(Request $request, \Closure $next): mixed
+    {
+        /** @var ?User $user */
+        $user = $request->user();
+
+        $uri = rtrim($request->getRequestUri(), '/') . '/';
+        $route = $request->route()->getName();
+
+        // Auth and profile endpoints should always be available
+        if (!$user || Str::startsWith($uri, ['/auth/', '/profile']) || Str::startsWith($route, ['auth.', 'account.', 'filament.app.auth.'])) {
```

Comment:
There's no route starting with `account.`.
> Also profile is registered under `filament.*.auth.profile` so there's no need for $uri

`routeIs` seems like a better fit
```suggestion
        // Auth and profile endpoints should always be available
        if (!$user || $request->routeIs('*auth.*')) {
```
Very permitive

```suggestion
        // Auth and profile endpoints should always be available
        if (!$user || $request->routeIs('*auth.*', 'filament.*.auth.*')) {
```
Only matches routes under `auth.*` & `filament.*.auth.*` so if you have `plugin.auth` it won't be granted.